src/modbus-reply.c

```diff
@@ -410,6 +410,21 @@ int modbus_reply_callback(modbus_t *ctx, const uint8_t *req, int req_length)
     }
 
 send_response:
+    /* Check if a user read/write callback returned a negated error code within the
+     * range of modbus protocol exceptions, if so respond with that exception */
```

Comment:
Alternatively could get the user to set `errno` in the callback, instead of returning the negated error number if you think that's better